src/main/kotlin/com/sourcegraph/cody/config/CodyAuthenticationManager.kt

```diff
@@ -129,14 +183,15 @@ class CodyAuthenticationManager(val project: Project) : Disposable {
     else null
   }
 
-  fun setActiveAccount(account: CodyAccount?) {
+  fun setActiveAccount(account: CodyAccount?, previousToken: String?) {
     if (!project.isDisposed) {
       val previousAccount = getActiveAccount()
-      val previousToken = previousAccount?.let { getTokenForAccount(it) }
```

Comment:
It is a possible and valid case to update the access token for an existing account. Previously we did not catch the access token change since at this point the access token is already updated. This is why I pass it as an argument.